src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -1453,10 +1453,15 @@ void GcsActorManager::RestartActor(const ActorID &actor_id,
       << ", death context type = " << GetActorDeathCauseString(death_cause)
       << ", remaining_restarts = " << remaining_restarts;
 
-  if (remaining_restarts != 0) {
+  if (remaining_restarts != 0 ||
+      (need_reschedule && max_restarts > 0 && mutable_actor_table_data->preempted())) {
     // num_restarts must be set before updating GCS, or num_restarts will be inconsistent
     // between memory cache and storage.
-    mutable_actor_table_data->set_num_restarts(num_restarts + 1);
+    if (mutable_actor_table_data->preempted()) {
+      mutable_actor_table_data->set_preempted(false);
+    } else {
+      mutable_actor_table_data->set_num_restarts(num_restarts + 1);
+    }
```

Comment:
please leave a comment here and in the actor table protobuf indicating that `num_restarts` does not include those due to pre-emption